components/brave_wallet_ui/page/container.tsx

```diff
@@ -365,10 +365,24 @@ function Container (props: Props) {
     }
   }
 
-  const onConnectHardwareWallet = (opts: HardwareWalletConnectOpts): Result.Type<HardwareWalletAccount[]> => {
-    // TODO (DOUGLAS): Add logic to connect a hardware wallet
+  const onConnectHardwareWallet = (opts: HardwareWalletConnectOpts): Promise<HardwareWalletAccount[]> => {
+    return new Promise(async (resolve, reject) => {
+      const keyring = await apiProxy.getInstance().getKeyringsByType(opts.hardware)
+      const controller = await apiProxy.getInstance().ethJsonRpcController
+      keyring.getAccounts(opts.startIndex, opts.stopIndex).then(async (accounts: HardwareWalletAccount[]) => {
+        for (let index = 0; index < accounts.length; index++) {
+          const balance = await controller.getBalance(accounts[index].address)
+          if (balance.success) {
+            accounts[index].balance = formatBalance(balance.balance, 18)
+          }
+        }
+        resolve(accounts)
+      }).catch(reject)
+    })
+  }
 
-    return []
+  const onAddHardwareAccounts = (selected: HardwareWalletAccount[]) => {
+    console.log(selected)
```

Comment:
is this logging supposed to be here?

Reply:
yes, this is fake method and will be filled by implementation in my next PR